### PR TITLE
Automatically clear stopwatch on merging a PR

### PR DESCRIPTION
**Pull request by adelowo**
Whenever a timer is created, it should be closed automatically whenever the PR is merged and/or an issue/PR is closed as per https://github.com/go-gitea/gitea/pull/4327#issuecomment-401553843 . This PR adds that functionality .

This uses `ctx.Flash` to show a one time message that the timer would be closed automatically as per   https://github.com/go-gitea/gitea/pull/4327#issuecomment-403326074 

screen shot 2018-07-14 at 17 36 53

### CODE REVIEW

**Comment by jonasfranz** [comment]
I think that it might be a little bit confusing if you stop a stopwatch without telling the user. I would propose to add a hint to the UI to indicate that the stopwatch will be stopped.

**Comment by adelowo** [comment]
Hi @JonasFranzDEV , where should that go in.. UI related things aren't what I would consider easy but could use some direction..


BTW, I was actually thinking it could more of a repo/global setting ???

**Comment by jonasfranz** [comment]
@adelowo Just a hint at the submit button with the info that it will stop the stopwatch.

**Comment by lafriks** [comment]
Wouldn't it be better in issue closing function. This way it would stop timer when closing issue or PR and also when merging I think

**Comment by adelowo** [comment]
Nice one @lafriks ... I would try that out

@JonasFranzDEV Would do just that

**Comment by adelowo** [comment]
Would update this PR over the weekend

**Comment by bkcsoft** [comment]
@JonasFranzDEV I think we could use `ctx.Flash` for this :thinking:

**Comment by adelowo** [comment]
Hi @lafriks , i tried that but merging a PR didn't trigger this when I placed it in the issue closing function.. I was able to do it some other way though .

File: options/locale/locale_en-US.ini [hunk 1]
```diff
@@ -753,6 +753,7 @@ issues.tracker = Time Tracker
 issues.start_tracking_short = Start
 issues.start_tracking = Start Time Tracking
 issues.start_tracking_history = `started working %s`
+issues.tracker_auto_close = Timer would be stopped automatically when this issue is closed
```
Comment (jonasfranz) [inline options/locale/locale_en-US.ini]: Use `Timer will be stopped automatically when this issue gets closed` instead

**Review by jonasfranz — DISMISSED** [review]

File: options/locale/locale_en-US.ini (hunk 1 above)
Comment (adelowo) [reply options/locale/locale_en-US.ini]: Done @JonasFranzDEV

**Comment by adelowo** [comment]
This is ready to be reviewed anyways ....

**Comment by jonasfranz** [comment]
Timer gets stopped if I reopen an issue. The message will be shown too if an issue is closed.

File: routers/user/auth.go [hunk 2]
```diff
@@ -1035,7 +1035,7 @@ func Activate(ctx *context.Context) {
 
 		ctx.Session.Set("uid", user.ID)
 		ctx.Session.Set("uname", user.Name)
-		ctx.Redirect(setting.AppSubURL + "/")
+		ctx.Redirect(setting.AppSubURL + "/user/login")
```
Comment (lafriks) [inline routers/user/auth.go]: This change does not seem related to this PR

Comment (adelowo) [reply routers/user/auth.go]: It somehow sneaked in from #4447

**Comment by techknowlogick** [comment]
@jonasfranz please review

**Review by techknowlogick — APPROVED** [review]

**Review by lafriks — APPROVED** [review]
